### PR TITLE
Add two new functions reboot and wait_reboot

### PR DESCRIPTION
**Pull request by player-29**
Added two new functions reboot and wait_reboot which is calling self.allow_restart_journal_messages and self.machine.reboot(timeout_sec=timeout_sec)

### CODE REVIEW

File: test/common/testlib.py [hunk 1]
```diff
@@ -2093,6 +2101,7 @@ def enable_root_login(self):
             self.sed_file('/root/d', disallowed_conf)
 
 
+
```
Comment (marusak) [inline test/common/testlib.py:2104]: Empty line, please drop

**Comment by marusak** [comment]
Thank you!

code wise just one blank forgotten line. @jelly is this what you had in mind?
It would be actually useful if you would use these new functions in tests. That would involve looking into `test/verify` where we call these functions and replace them with these new helpers while also removing the `allow_restart_journal_messages()` calls.

Also please update your commit message. Good practice is to look in some commit messages for the same files (`git log test/common/testlib.py`) to see how team usually writes these and also very good article to read is https://cbea.ms/git-commit/ . Please also don't forget to link the issue in the commit message, see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by jelly** [comment]
> Thank you!
> 
> code wise just one blank forgotten line. @jelly is this what you had in mind? It would be actually useful if you would use these new functions in tests. That would involve looking into `test/verify` where we call these functions and replace them with these new helpers while also removing the `allow_restart_journal_messages()` calls.

Yes, that is what I had in mind, it should happen in this PR so we can actually see that the new functions do what they intend to achieve :)

> Also please update your commit message. Good practice is to look in some commit messages for the same files (`git log test/common/testlib.py`) to see how team usually writes these and also very good article to read is https://cbea.ms/git-commit/ . Please also don't forget to link the issue in the commit message, see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: test/common/testlib.py [hunk 2]
```diff
@@ -2091,8 +2099,6 @@ def enable_root_login(self):
         disallowed_conf = '/etc/cockpit/disallowed-users'
         if not self.machine.ostree_image and self.file_exists(disallowed_conf):
             self.sed_file('/root/d', disallowed_conf)
-
-
```
Comment (jelly) [inline test/common/testlib.py:2095]: Please don't drop the whitelines here.

File: test/common/testlib.py [hunk 3]
```diff
@@ -2080,6 +2080,14 @@ def write_file(self, path: str, content: str, append: bool = False, owner: Optio
...
         self.restore_file(path, post_restore_action=post_restore_action)
         m.write(path, content, append=append, owner=owner, perm=perm)
+    
+    def reboot(self, timeout_sec):
+        self.allow_restart_journal_messages()
+        self.machine.reboot(timeout_sec=timeout_sec)
+
+    def wait_reboot(self, timeout_sec):
+        self.allow_restart_journal_messages()
+        self.machine.wait_reboot(timeout_sec=timeout_sec)
```
Comment (jelly) [inline test/common/testlib.py:2090]: This is great, but nothing actually uses it. Please adjust the existing tests to use the new `self.wait_reboot`. For example `testTracer` in `test/verify/check-packagekit`.

* Adjust the usage of `m.wait_reboot` in `test/verify/check-packagekit`
* Prepare a test virtual machine: `./test/image-prepare -q fedora-37`
* Run the test: `TEST_OS=fedora-37 ./test/verify/check-packagekit -vst`

If this works as expected (test passes), adjust all other occurrences of `m.wait_reboot` in `test/verify`.

**Review by jelly — CHANGES_REQUESTED** [review]

File: test/common/testlib.py [hunk 4]
```diff
@@ -2099,6 +2099,8 @@ def enable_root_login(self):
         disallowed_conf = '/etc/cockpit/disallowed-users'
         if not self.machine.ostree_image and self.file_exists(disallowed_conf):
             self.sed_file('/root/d', disallowed_conf)
+
```
Comment (marusak) [inline test/common/testlib.py:2102]: ` Add white lines ` Why? :) Is this some linter that you use?

Comment (player-29) [reply test/common/testlib.py:2102]: Nothing like that, but @jelly instructed me not to drop the white lines there.

**Comment by KKoukiou** [comment]
@player-29 you 're very close to get this ready! Did you see the comments from https://github.com/cockpit-project/cockpit/pull/18431#discussion_r1122949054?

**Comment by player-29** [comment]
> @player-29 you're very close to get this ready! Did you see the comments from [#18431 (comment)](https://github.com/cockpit-project/cockpit/pull/18431#discussion_r1122949054)?
thank you for reviewing it. Yes, I have added the code to existing tests to use the new functions but facing an issue in testing it.

**Comment by KKoukiou** [comment]
closing because of inactivity.

**Pull request closed without merging**